src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -105,7 +110,9 @@ public PerTargetPropagater(LinkedPointSet targets, AnalysisTask task,
         // If we're making a static site we'll break travel times down into components and make paths.
         // This expects the pathsToStopsForIteration and pathWriter fields to be set separately by the caller.
         this.calculateComponents = task.makeStaticSite;
-        speedMillimetersPerSecond = (int) (request.walkSpeed * 1000);
+        StreetMode egressMode = LegMode.getDominantStreetMode(task.egressModes);
+        speedMillimetersPerSecond = (int) (request.getSpeedForMode(egressMode) * MM_PER_METER);
+        egressLegTimeLimitSeconds = (int) request.getMaxAccessTimeForMode(egressMode) * SECONDS_PER_MINUTE;
```

Comment:
If both multiplication operands are ints the result will be an int. If at least one of them is floating point the result will be floating point. request.getSpeedForMode returns a float, the result of the multiplication is a float, and you need an explicit cast to truncate it for storage in an integer variable. request.getMaxAccessTimeForMode returns an int though, so the result of the multiplication is an int and the cast is redundant. An IDE should show you that it's redundant. However also be aware that the operator precedence of casting is higher than multiplication or division, so here you're actually casting the output of getMaxAccessTimeForMode (which is already an int) to an int before multiplying it. If getMaxAccessTimeForMode was a method returning a floating point value, this would truncate the value before the multiplication, which would be bad.